### ISSUE TITLE
MA0040 in expressions that return CancellationToken

### ISSUE THREAD

**Issue by hankovich** [meziantou/Meziantou.Analyzer#586]
### Version of the Meziantou.Analyzer NuGet package

2.0.74

### Rule Identifier

MA0040

### Target Framework

net8.0

### C# Language version

C# 12

### Description

MA0040 is triggered in expressions that itself return `CancellationToken`

### Reproduction Steps

Minimal code:

```c#
CancellationToken Foo(CancellationToken cancellationToken = default) => cancellationToken;

var token = Foo(); // Use an overload with a CancellationToken, available tokens: token
```

**Comment by meziantou** [comment]
It will be fixed in the next version!
